async/unstable_wait_for_test.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import { assertAlmostEquals, assertEquals, assertRejects } from "@std/assert";
+import { waitFor } from "./unstable_wait_for.ts";
+
+// NOT detecting leaks means that the internal interval was correctly cleared
+
+Deno.test("waitFor() returns fulfilled promise", async () => {
+  let flag = false;
+  setTimeout(() => flag = true, 100);
+  const start = Date.now();
+  await waitFor(() => flag === true, 1000);
+  // Expects the promise to be resolved after 100ms
+  assertAlmostEquals(Date.now() - start, 100, 10);
+});
+
+Deno.test("waitFor() throws DOMException on timeout", async () => {
+  let flag = false;
+  setTimeout(() => flag = true, 1000);
```

Comment:
This timer needs to be cleared at the end of the test case to avoid op leak error

Reply:
Will do, makes sense. I do not know why I did not get the leak when running locally.